### PR TITLE
Update to @syntest/core@0.3.1

### LINKED ISSUE #152: Upgrade to @syntest/core:0.3.1

### PR DESCRIPTION
**Pull request by dstallenberg**
Depends on the release of @syntest/core@0.3.1

Closes #152

### CODE REVIEW

File: src/util/collection.ts [hunk 1]
```diff
@@ -42,19 +42,19 @@ export function collectInitialVariables<T extends Encoding>(
...
   collector.recordVariable(RuntimeVariable.SUBJECT, path.basename(targetPath));
   collector.recordVariable(
     RuntimeVariable.PROBE_ENABLED,
-    `${Properties.probe_objective}`
+    `${CONFIG.probeObjective}`
   );
   collector.recordVariable(
     RuntimeVariable.CONSTANT_POOL_ENABLED,
-    `${Properties.constant_pool}`
+    `${CONFIG.constant_pool}`
```
Comment (mitchellolsthoorn) [inline src/util/collection.ts:55]:
```suggestion
    `${CONFIG.constantPool}`
```

The naming is not consistent

File: src/testcase/sampling/SolidityRandomSampler.ts [hunk 2]
```diff
@@ -164,7 +165,7 @@ export class SolidityRandomSampler extends SoliditySampler {
       (<SoliditySubject>this._subject)
         .getPossibleActions()
         .filter((a) => a.type === type.type).length &&
-      prng.nextBoolean(Properties.sample_func_as_arg)
+      prng.nextBoolean(CONFIG.sampleFunctionOutputAsArgument)
```
Comment (mitchellolsthoorn) [inline src/testcase/sampling/SolidityRandomSampler.ts:168]: This property was renamed?

File: src/SolidityLauncher.ts [hunk 3]
```diff
@@ -496,23 +460,31 @@ export class SolidityLauncher {
...
+        crossover
+        );
 
-      await suiteBuilder.clearDirectory(Properties.temp_test_directory);
+      await suiteBuilder.clearDirectory(CONFIG.tempTestDirectory);
 
       // allocate budget manager
-      const iterationBudget = new IterationBudget(Properties.iteration_budget);
+      const iterationBudget = new IterationBudget(CONFIG.iterationBudget);
       const evaluationBudget = new EvaluationBudget();
```
Comment (mitchellolsthoorn) [inline src/SolidityLauncher.ts:475]: We don't have a config for the evaluation budget?

**Review by mitchellolsthoorn — CHANGES_REQUESTED** [review]

File: src/util/collection.ts (hunk 1 above)
Comment (dstallenberg) [reply src/util/collection.ts:55]: Fixed

File: src/testcase/sampling/SolidityRandomSampler.ts (hunk 2 above)
Comment (dstallenberg) [reply src/testcase/sampling/SolidityRandomSampler.ts:168]: Yes

File: src/SolidityLauncher.ts (hunk 3 above)
Comment (dstallenberg) [reply src/SolidityLauncher.ts:475]: We do, but this was how i found it. Added the config now.

File: src/SolidityLauncher.ts [hunk 4]
```diff
@@ -87,114 +86,128 @@ import {
   collectInitialVariables,
   collectStatistics,
 } from "./util/collection";
+import Yargs = require("yargs");
+import { RemoveIndex } from "@syntest/core/dist/Configuration";
```
Comment (mitchellolsthoorn) [inline src/SolidityLauncher.ts:90]: We should not refer to the dist folder. This is making the build fail as well

**Review by mitchellolsthoorn — CHANGES_REQUESTED** [review]

File: src/SolidityLauncher.ts (hunk 3 above)
Comment (mitchellolsthoorn) [reply src/SolidityLauncher.ts]: I don't see the fix in the PR

Comment (dstallenberg) [reply src/SolidityLauncher.ts]: Weird I dont see it either in the PR but if you look in the file on the branch it is there.

File: src/SolidityLauncher.ts (hunk 4 above)
Comment (dstallenberg) [reply src/SolidityLauncher.ts]: Fixed, but the build will keep failing until we release a new beta of the core and update the dependency in this project.

**Pull request closed without merging**